Towny/src/main/java/com/palmergames/bukkit/towny/object/Resident.java

```diff
@@ -321,6 +315,13 @@ public void removeTown() {
 			}
 		}
 		
+		BukkitTools.fireEvent(new TownRemoveResidentEvent(this, town));
+
+		try {
+			town.removeResident(this);
+		} catch (NotRegisteredException ignored) {} catch (EmptyTownException e) {
```

Comment:
Flipping the order of the catches and putting the NRE catch on a newline would look a bit better